### ISSUE TITLE
Key UNRESOLVED found in VariantContext field FILTER

### ISSUE THREAD

**Issue by CristianRiccio** [fritzsedlazeck/Sniffles#82]
Hi,

I have called SVs using Sniffles. I then tried to sort the VCF file using `picard SortVcf` but I got the following error:

`Key UNRESOLVED found in VariantContext field FILTER at I:2146719 but this key isn't defined in the VCFHeader.  We require all VCFs to have complete VCF headers by default.` 

What does 'UNRESOLVED' mean? It comes as a FILTER with some insertions. Does it mean Sniffles is not sure about the insertion?

Also, would you be able to update the code and define UNRESOLVED in the VCF header?

Cheers.

**Comment by CristianRiccio** [comment]
My VCF contains 

`##INFO=<ID=UNRESOLVED,Number=0,Type=Flag,Description="An insertion that is longer than the read and thus we cannot predict the full size.">` 

but `picard SortVcf` still complains:

`Exception in thread "main" java.lang.IllegalStateException: Key UNRESOLVED found in VariantContext field FILTER at I:2146719 but this key isn't defined in the VCFHeader.  We require all VCFs to have complete VCF headers by default.`

I am using version 1.0.8. Is there a newer version that defines this 'UNRESOLVED' differently?

**Comment by fritzsedlazeck** [comment]
Thanks for pointing this out. I will check this week.
Cheers
Fritz

**Comment by fritzsedlazeck** [comment]
So I just pushed a code since I forgot a different flag to define. So please clone the code and try the newest version. 

I ran the picard tools sortVCF and it ran through on a toy example. Can you maybe retry?
If the error persists can you maybe share the VCF file? Either direct or over email: fritz.sedlazeck@gmail.com

Thanks
Fritz

**Comment by CristianRiccio** [comment]
Hi,

I am now using version 1.0.7 and I still have the same problem. However, the header does contain the UNRESOLVED definition. Maybe there is a format problem? I've sent you the file by email.

**Comment by fritzsedlazeck** [comment]
Hi Christian,
can you describe the error/ problem? 
Thanks
Fritz

**Comment by CristianRiccio** [comment]
Exception in thread "main" java.lang.IllegalStateException: Key
UNRESOLVED found in VariantContext field FILTER at chrI:265864 but this
key isn't defined in the VCFHeader. We require all VCFs to have complete
VCF headers by default. 

Il 2019-02-21 17:35 Fritz Sedlazeck ha scritto: 

> Hi Christian,
> can you describe the error/ problem?
> Thanks
> Fritz 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/fritzsedlazeck/Sniffles/issues/82#issuecomment-466092594
[2]

**Comment by fritzsedlazeck** [comment]
Can you also let me know which tool it is?

**Comment by CristianRiccio** [comment]
picard SortVcf 

picard SortVcf
INPUT=output/variants/calling/reads/ngmlr/sniffles/ref/vc2010/addisababa/renamed.vcf.gz
OUTPUT=output/variants/calling/reads/ngmlr/sniffles/ref/vc2010/addisababa/renamedSorted.vcf.gz
SEQUENCE_DICTIONARY=output/genome/celegans/ref/vc2010/index/picard/vc2010renamed.dict


Il 2019-02-21 17:47 Fritz Sedlazeck ha scritto: 

> Can you also let me know which tool it is? 
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or mute the thread [2].
 

Links:
------
[1]
https://github.com/fritzsedlazeck/Sniffles/issues/82#issuecomment-466096881
[2]

**Comment by CristianRiccio** [comment]
picard SortVcf

**Comment by CristianRiccio** [comment]
actually, I am using version 1.0.11 on my linux machine but I get the same problem

**Comment by fritzsedlazeck** [comment]
Yeah it has to do with Unresolved being defined as INFO Tag and not FILTER. 
i am trying to figure it out... VCF is sometimes iffy for the header definitions. 

If you need a faster workaround I usually use bedtools sort. which works.
Cheers
Fritz

**Comment by CristianRiccio** [comment]
bedtools sort -i renamed.vcf.gz is also complaining.

Error: malformed VCF entry at line 3542. Start was greater than end. Exiting.

I don't understand why start cannot be greather than end. In the case of translocations, the start on one chromosome could be after the end on another chromosome.

**Comment by fritzsedlazeck** [comment]
Ok i just pushed a fix that works for me on picard tools. 
Can you clone the source and rerun?
Thanks
Fritz

**Comment by MarioWassmer** [comment]
Hi Fritz. I discovered the same issue today. The VCF contains two insertions with a length of approx. 1033 bp that are represented in the GIAB reference as well. Both contains the string UNRESOLVED in the Filter column and the length is set to 999999999 in the file.
The vcf was produced with SNIFFLES 10.0.11 after minimap2 alignment of nanopore reads to human hs37d5.

Mario

**Comment by fritzsedlazeck** [comment]
Hi Mario,
this is expected. When Sniffles can detect that there is an insertion but unclear what the length is it reports it as unresolved. Given the VCF standard, we need to put in some length of the SV and thus I just report the huge number. 
sorry for the confusion.
Fritz
